src/tool/xmeta/cpp/models/class_model.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "base_model.h"
+#include "interface_model.h"
+#include "method_model.h"
+#include "property_model.h"
+#include "event_model.h"
+
+namespace xlang
+{
+    namespace xmeta
+    {
+        // Note: All classes are implicitly public, and explicitly specifying this is not allowed.
+        enum class_modifier_t
+        {
+            none,
+            sealed_class,
+            static_class
+        };
+
+        class class_model final : public base_model
+        {
+        public:
+            class_model(const std::string &id, const size_t &decl_line) : base_model{ id, decl_line } { }
+
+            std::string class_base_id;
+            std::shared_ptr<class_model> class_base;
+            std::vector<std::string> interface_base_list;
```

Comment:
consider using plural forms instead of `_list` as a suffix, `interface_bases`, etc.